newIDE/app/src/EventsFunctionsExtensionEditor/index.js

```diff
@@ -266,6 +267,11 @@ export default class EventsFunctionsExtensionEditor extends React.Component<
       return;
     }
 
+    // Users may have change a function declaration.
+    // Reload metadata just in case.
+    if (this.props.onFunctionEdited)
+      this.props.onFunctionEdited(this.props.eventsFunctionsExtension);
```

Comment:
Should we await this async function?
It could avoid strange issues but make the UI a bit slower.

Reply:
Risk an inconsistent UI => or worse! If something is awaiting something else, the "something" (the extension/project...) can not even exist anymore if in the meantime another action was taken (extension deleted, project closed).
So let's ensure the state is always updated immediately, and we can talk about making this visible to the user later (the best would even be some worker doing this in the background, but hard to do currently)